src/services/Client.ts

```diff
@@ -4,42 +4,73 @@ import Consumer from './Consumer';
 import Producer from './Producer';
 import { RTC_CONFIG } from '../constants';
 
-type HandshakeProps = {
-  clientId: string;
+// type HandshakeProps = {
+//   receiver: string;
+//   remotePeerId: string;
+//   description: RTCSessionDescription;
+//   candidate: RTCIceCandidate;
+// };
+
+type HandshakeData = {
+  sender: string;
+  receiver: string;
   remotePeerId: string;
   description: RTCSessionDescription;
   candidate: RTCIceCandidate;
 };
 
 class Client {
   id: string;
-
   socket: typeof socket;
-
   producer: Producer;
-
   consumers: Map<string, Consumer>;
 
   onUpdateConsumers: (consumers: Map<string, Consumer>) => void;
 
   constructor(onUpdateConsumers: (consumers: Map<string, Consumer>) => void) {
     this.id = uuidv4();
     console.log(`%cI AM CLIENT ${this.id}`, 'color: green');
-    this.socket = socket;
+    this.socket = null;
     this.producer = new Producer(
       this.socket,
       this.id,
       RTC_CONFIG,
-      this.updateFeatures.bind(this),
+      this.updateFeatures.bind(this)
     );
     this.consumers = new Map();
     this.onUpdateConsumers = onUpdateConsumers;
-    this.bindSocketEvents();
+  }
+
+  createWebSocket() {
+    this.socket = new WebSocket('wss://rufdlv7k6k.execute-api.us-east-2.amazonaws.com/production');
```

Comment:
like with the SFU, we should consider how we incorporate this url for long-term maintanability